src/tail/mod.rs

```diff
@@ -19,12 +20,27 @@ impl Tail {
         let mut runtime = TokioRuntime::new()?;
 
         runtime.block_on(async {
+            // Create three one-shot channels for handling ctrl-c
+            let (log_tx, log_rx) = oneshot::channel();
+            let (session_tx, session_rx) = tokio::sync::oneshot::channel();
+            let (tunnel_tx, tunnel_rx) = tokio::sync::oneshot::channel();
+
+            // Pass the three transmitters to a newly spawned sigint handler
+            let txs = vec![log_tx, tunnel_tx, session_tx];
+            let sigint_handle = tokio::spawn(handle_sigint(txs));
```

Comment:
```suggestion
            let sigint_handler = tokio::spawn(handle_sigint(txs));
```

Reply:
technically these are all `JoinHandle`s but i also sort of hate calling things by their mechanical names...